### PR TITLE
created poll API's

### PR DESCRIPTION
**Pull request by deepakr123**

### CODE REVIEW

File: controllers/v1/mediaFilesController.js [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+    constructor() {
+        super(mediaFilesSchema);
+    }
+
+    static get name() {
+        return "mediaFiles";
+    }
+
+     /**
+     * @api {post} /assessment/api/v1/mediaFiles/createEmoji Create Emoji
```
[the remark this answers is not in the document]
Comment (deepakr123) [reply controllers/v1/mediaFilesController.js:27]: resolved

File: controllers/v1/mediaFilesController.js [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+                status: error.status || httpStatusCode.internal_server_error.status,
+                message: error.message || httpStatusCode.internal_server_error.message,
+                errorObject: error
+            });
+        }
+    })
+}
+
+
+    /**
```
[the remark this answers is not in the document]
Comment (deepakr123) [reply controllers/v1/mediaFilesController.js:144]: resolved

File: controllers/v1/unicodesController.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ * author : Deepa
+ * created-date : 17-Aug-2020
+ * Description : unicodes related information.
+ */
+
+/**
+    * Unicodes
+    * @class
+*/
+module.exports = class Unicodes extends Abstract {
```
[the remark this answers is not in the document]
Comment (deepakr123) [reply controllers/v1/unicodesController.js]: resolved

File: models/polls.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+module.exports = {
+  name: "polls",
+  schema: {
+    name: String,
+    creator: {
```
[the remark this answers is not in the document]
Comment (deepakr123) [reply models/polls.js]: resolved

File: models/polls.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+      type: String,
+      required: true,
+      index: true
+    },
+    questions: Array,
+    link: String,
```
[the remark this answers is not in the document]
Comment (deepakr123) [reply models/polls.js]: resolved

File: generics/helpers/utils.js [hunk 6]
```diff
@@ -1,3 +1,5 @@
+const uuid = require('uuid/v1');
```
Comment (aks30) [inline generics/helpers/utils.js:1]: @deepakr123 - Use v4

File: generics/helpers/utils.js [hunk 7]
```diff
@@ -227,5 +240,6 @@ module.exports = {
   checkIfEnvDataExistsOrNot : checkIfEnvDataExistsOrNot,
   fetchAssessorsLeadAssessorRole : fetchAssessorsLeadAssessorRole,
   epochTime : epochTime,
-  isValidMongoId : isValidMongoId
+  isValidMongoId : isValidMongoId,
+  uuidGenerate : uuidGenerate
```
Comment (aks30) [inline generics/helpers/utils.js:244]: @deepakr123 - Rename it generateUUID

File: generics/middleware/authenticator.js [hunk 8]
```diff
@@ -115,6 +115,13 @@ module.exports = async function (req, res, next) {
     return res.status(401).send(respUtil(rspObj));
   }
 
+   if ((req.path.includes("createGesture")) || (req.path.includes("createEmoji")) && (req.headers["internal-access-token"] !== process.env.INTERNAL_ACCESS_TOKEN)) {
```
Comment (aks30) [inline generics/middleware/authenticator.js:118]: @deepakr123 - Refer to sunbird service implementation as - https://github.com/shikshalokam/sl-sunbird-service/blob/master/generics/middleware/authenticator.js

File: models/mediaFiles.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
+module.exports = {
+    name: "mediaFiles",
+    schema: {
+        name: {
+          type: String,
+          required: true,
+          index: true
+        },
+        type: {
```
Comment (aks30) [inline models/mediaFiles.js:9]: @deepakr123 - Put type, Unicode, and status in an index.

File: models/polls.js [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+      index: true
+    },
+    questions: Array,
+    link: {
+      type: String,
+      index: true
+    },
+    result: Object,
+    numberOfResponses: Number,
+    metaInformation: Object,
```
Comment (aks30) [inline models/polls.js:18]: @deepakr123 - Please add the logic of putting all the form fields which are not there in the polls model in the metaInformation

File: models/polls.js [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+    isDeleted: {
+      type: Boolean,
+      default: false
+    },
+    startDate: Date,
```
Comment (aks30) [inline models/polls.js:23]: @deepakr123 - Make fields which will always be there as mandatory.

File: module/pollSubmissions/helper.js [hunk 12]
```diff
@@ -0,0 +1,181 @@
...
+
+                if (Object.keys(responseObject).length == 0) {
+                    throw new Error (messageConstants.apiResponses.RESPONSE_OBJECT_REQUIRED_CHECK)
+                }
+
+                if (userId == "") {
+                    throw new Error(messageConstants.apiResponses.USER_ID_REQUIRED_CHECK)
+                }
+
+                let pollSubmissionDocument = await this.pollSubmissionDocuments
```
Comment (aks30) [inline module/pollSubmissions/helper.js:92]: @deepakr123 - Put an exist true condition here.

File: module/pollSubmissions/helper.js [hunk 13]
```diff
@@ -0,0 +1,181 @@
...
+                        if (!result[singleResponse.qid].responses[singleResponse.label]) {
+                            result[singleResponse.qid].responses[singleResponse.label] = 1;
+                        }
+                        else {
+                            result[singleResponse.qid].responses[singleResponse.label] = result[singleResponse.qid].responses[singleResponse.label] + 1;
+                        }
+                    }
+                })
+
+                await database.models.polls.updateOne
```
Comment (aks30) [inline module/pollSubmissions/helper.js:158]: @deepakr123 - Use helper function to update poll results.

File: module/polls/helper.js [hunk 14]
```diff
@@ -0,0 +1,526 @@
...
+                    },
+                    [
+                      "value"
+                    ]
+                )
+
+                pollCreationForm = [...pollCreationForm[0].value,
+                                    ...pollCreationForm[1].value]
+                
+                let unicodes = await database.models.mediaFiles.find
```
Comment (aks30) [inline module/polls/helper.js:96]: @deepakr123 - Use module helper functions

File: module/polls/helper.js [hunk 15]
```diff
@@ -0,0 +1,526 @@
...
+                    question.options = options;
+                    questionArray.push(question);
+                });
+                
+                pollDocument.questions = questionArray;
+                
+                let createPollResult = await database.models.polls.create(pollDocument)
+
+                let uuid = gen.utils.uuidGenerate();
+                let link = md5(uuid + "###" + createPollResult._id);
```
Comment (aks30) [inline module/polls/helper.js:198]: @deepakr123 - Create utils function for md5

File: models/mediaFiles.js [hunk 16]
```diff
@@ -0,0 +1,19 @@
+module.exports = {
+    name: "mediaFiles",
+    schema: {
+        name: String,
+        type: String,
```
[the remark this answers is not in the document]
Comment (deepakr123) [reply models/mediaFiles.js]: resolved

File: generics/helpers/utils.js (hunk 7 above)
Comment (deepakr123) [reply generics/helpers/utils.js:244]: resolved

File: generics/helpers/utils.js (hunk 6 above)
Comment (deepakr123) [reply generics/helpers/utils.js:1]: resolved

Comment (deepakr123) [reply generics/helpers/utils.js:1]: resolved

File: module/polls/helper.js (hunk 15 above)
Comment (deepakr123) [reply module/polls/helper.js:198]: resolved

File: models/mediaFiles.js (hunk 16 above)
[the remark this answers is not in the document]
Comment (deepakr123) [reply models/mediaFiles.js]: resolved

File: models/mediaFiles.js (hunk 9 above)
Comment (deepakr123) [reply models/mediaFiles.js:9]: resolved

File: module/pollSubmissions/helper.js (hunk 13 above)
Comment (deepakr123) [reply module/pollSubmissions/helper.js:158]: resolved

File: module/polls/helper.js (hunk 14 above)
Comment (deepakr123) [reply module/polls/helper.js:96]: resolved

File: models/polls.js (hunk 11 above)
Comment (deepakr123) [reply models/polls.js:23]: resolved

File: module/pollSubmissions/helper.js (hunk 12 above)
Comment (deepakr123) [reply module/pollSubmissions/helper.js:92]: resolved

Comment (deepakr123) [reply module/pollSubmissions/helper.js:92]: resolved

File: models/polls.js (hunk 10 above)
Comment (deepakr123) [reply models/polls.js:18]: resolved

**Pull request closed without merging**